src/arrayMethodJoin.js

```diff
@@ -5,7 +5,25 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    const actualSeparator = separator === undefined ? ',' : separator;
+
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      const element = this[i];
+
+      if (i > 0) {
+        result += actualSeparator;
```

Comment:
The separator should not be appended if the current element is the last one in the array. You should check if i < this.length - 1 before appending the separator.